src/ui/qgsmapsavedialog.ui

```diff
@@ -16,6 +16,19 @@
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QGridLayout" name="gridLayout">
+     <item row="8" column="0" colspan="2">
+      <widget class="QCheckBox" name="mSaveAsRaster">
+       <property name="text">
+        <string>Save as raster</string>
```

Comment:
I think this could be better named, e.g. "rasterise map when exporting" with a tooltip explaining why  you'd want this option